### PR TITLE
Add new board: Purism Librem Mini

### PR DESCRIPTION
**Pull request by MrChromebox**
Add support for the Purism Librem Mini:

* Add coreboot 4.12 patch to add board support
* Add coreboot 4.12 patch to show SoC ME status on disabled ME
* Add board and coreboot configs
* Add purism-blobs coreboot submodule (vs using get_blobs script)
* Adjust librem linux config for newer 4.19 kernel
* Rename all librem boards and configs for consistency

Depends on #805 / will be rebased after merge

### CODE REVIEW

**Comment by tlaurion** [comment]
Will depend on build success of https://app.circleci.com/pipelines/github/tlaurion/heads/334/workflows/5eb74b45-46db-4f13-b789-803237929377/jobs/362

@MrChromebox please rebase

**Comment by MrChromebox** [comment]
@tlaurion done

File: modules/coreboot [hunk 1]
```diff
@@ -8,6 +8,7 @@ else ifeq "$(CONFIG_COREBOOT_VERSION)" "4.12"
 	coreboot_version := 4.12
 	coreboot_hash := edcad000ee9b73183c396ea76155629b3d27c693e0f1ae83e3424c4d936e2be2
 	coreboot-blobs_hash := 4735ee6850d55d1e65dee8b08cc9b28b8af00b42acf31365f5d9545406579104
+	coreboot_depends := $(if $(CONFIG_PURISM_BLOBS), purism-blobs)
```
Comment (tlaurion) [inline modules/coreboot:11]: @MrChromebox We can chainload multiple if here for different board configurations?

```
coreboot_depends := $(if $(CONFIG_PURISM_BLOBS), purism-blobs)
coreboot_depends := $(if $(CONFIG_xx30_BLOBS),xx30-blobs)
```

etc?

Comment (MrChromebox) [reply modules/coreboot:11]: > @MrChromebox We can chainload multiple if here for different board configurations?

would need to be a single line I believe:
`coreboot_depends := $(if $(CONFIG_PURISM_BLOBS), purism-blobs) $(if $(CONFIG_xx30_BLOBS),xx30-blobs)`

**Comment by tlaurion** [comment]
@MrChromebox : consider adding https://github.com/tlaurion/heads/commit/12711eea6c39e9894748dc0b7e7c0e0a0f0b3910 and gitlabci counterpart prior of merging this PR if successful.

Building here, wont be able to test outcome https://circleci.com/gh/tlaurion/heads/377?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by tlaurion** [comment]
@MrChromebox build fails on docker debian:10 with 
```
Unpacked and patched ... ok
Building packages ...
Building GMP v6.1.2 for host ... ok
Building MPFR v4.0.2 for host ... ok
Building MPC v1.1.0 for host ... ok
Building BINUTILS v2.33.1 for target ... ok
Building GCC v8.3.0 for target ... failed. Check 'build-i386-elf-GCC/build.log'.
make[3]: *** [Makefile:26: build_gcc] Error 1
make[2]: *** [Makefile:51: build-i386] Error 2
make[1]: *** [util/crossgcc/Makefile.inc:35: crossgcc-i386] Error 2
make[1]: Leaving directory '/root/project/build/coreboot-4.12'
make: *** [modules/coreboot:70: "/root/project/build/coreboot-4.12/.xcompile"] Error 2
```
Src https://app.circleci.com/pipelines/github/tlaurion/heads/349/workflows/7330f8e8-4b0a-4540-862b-da021bc0f8aa/jobs/377

**Comment by MrChromebox** [comment]
@tlaurion issue building older GCC for coreboot toolchain I'm guessing

**Comment by tlaurion** [comment]
@MrChromebox Well, past CI against bullseye worked before for coreboot 4.12 until.a change in host build toolchain broke it.

CI was then changed to use Debian 10 docker image which works for coreboot 4.8.1 but  doesn't for coreboot 4.12.

A bit confused here. How can we make coreboot depend on musl-cross-make at this point... 

But then

**Comment by MrChromebox** [comment]
@tlaurion oops, missed that, I will CP and push to re-test once this build is done

**Comment by tlaurion** [comment]
@MrChromebox found anything related to the present problem?

status docker debian:bullseye:
- coreboot 4.12 builds, coreboot 4.8.1 fails

status docker debian:10:
- coreboot 4.8.1 builds, coreboot 4.12 fails.

**Comment by MrChromebox** [comment]
@tlaurion I'm not even sure how to begin diagnosing the issue. There don't seem to be any logs of the build to show what failed

**Comment by tlaurion** [comment]
@MrChromebox Will try to bring back what I implemented before to output content of last files modified.

**Comment by tlaurion** [comment]
@MrChromebox : might now be enough and we might want to replace tail from a full output of those logs, but you can get the idea here: https://app.circleci.com/pipelines/github/tlaurion/heads/364/workflows/df9bad8d-8ff1-40da-b8d8-1b87a05be509/jobs/392

**Comment by tlaurion** [comment]
@MrChromebox : https://app.circleci.com/pipelines/github/tlaurion/heads/364/workflows/df9bad8d-8ff1-40da-b8d8-1b87a05be509/jobs/392 was successful?!? Does the artifact boot for you?

**Comment by MrChromebox** [comment]
> @MrChromebox : https://app.circleci.com/pipelines/github/tlaurion/heads/364/workflows/df9bad8d-8ff1-40da-b8d8-1b87a05be509/jobs/392 was successful?!? Does the artifact boot for you?

yes, it boots successfully :)

**Comment by tlaurion** [comment]
@MrChromebox Will have to revert. Some patches in CircleCI will have to be done to output the logs of binutiles which fails at https://app.circleci.com/pipelines/github/tlaurion/heads/365/workflows/c32c3e29-f383-43b1-9ffe-5b5968c7feaf/jobs/393 :
```
Building GCC v8.3.0 for target ... failed. Check 'build-i386-elf-GCC/build.log'.
make[3]: *** [Makefile:26: build_gcc] Error 1
make[2]: *** [Makefile:51: build-i386] Error 2
make[1]: *** [util/crossgcc/Makefile.inc:35: crossgcc-i386] Error 2
make[1]: Leaving directory '/root/project/build/coreboot-4.12'
make: *** [modules/coreboot:70: "/root/project/build/coreboot-4.12/.xcompile"] Error 2
```
So https://github.com/osresearch/heads/blob/master/modules/coreboot#L68-L71 is the culprit, where `build-i386-elf-GCC/build.log` would be required to debug this further.

**Pull request merged**